### DISCUSSION TITLE
issue with grabber in Armbian hyperhdr config

### DISCUSSION THREAD

**Discussion by NiiroX** [awawa-dev/HyperHDR#290] [Q&A]
Hello
I use hyperhdr with an odroid C2 under armbian (Jammy then Bulleyes)
I have a problem with the colors of my leds (sk6812 RGBCW) which are inverted or totally fanciful, I modified the RGB on hyperhdr and wled but the problem is still there. The preview in hyperhdr also shows fancy color inversions. Using the grabber on windows 10 I don’t have that worry.
I forgot to mention that my grabber is a Rullz
Is it possible that under Armbian I encounter a driver problem? (grabber) If so, is it possible to solve the problem? 
Thank you

**Comment by awawa-dev** [comment]
Hi
There are likely 2 sources of the wrong color problem in video preview:  
1) Driver problem I encountered on one rk3328 device. The colors were not swapped but the image was completely unreadable. it only showed up when Rullz was plugged into USB2.0 port, it worked fine when plugged into USB3.0. The reason was an incorrect driver that reads brightness / saturation / hue incorrectly. But it could have been corrected by manually setting them in HyperHDR. I described it here: https://www.hyperhdr.eu/2022/03/tv-box-mania-iii-part-h96-max-plus.html 

2) The little / big endian operating system settings. HyperHDR due to the significant optimization of video processing is adapted to the little endian which this setting is standard for Intel and ARM processors. But almost all ARM processors can run on big endian as well, as long as the custom operating system forces it. The effect of this will be, for example: the swapped colors due to different arrangement of RGB colors in the memory. I only encountered this on RPi on one occasion when installing clean Debian for Rpi (not Raspian or Armbian). Because this is such a niche case, additionally dependent on the non-standard configuration of the operating system and I do not treat it as a bug and I do not envisage creating separate paths for video processing in the system for big endian, although of course it is completely doable.
To confirm it execute ```lscpu | grep Endian```

**Reply by NiiroX** [reply]
hi
Thank you for your help and for your incredible work!!!  The second solution doesn’t seem to work for me (tested on bulleyes but not on jammy)
Concerning the second solution I will test tonight the settings
Thanks again for your availability and the availability of your knowledge
Respect

**Comment by awawa-dev** [comment]
Thank you. Did you read your default settings for brightness/contrast/hue/saturation like on the screenshot in the blog article?

**Reply by NiiroX** [reply]
Yes I have this default settings:
Brightness : -8193
Contrast : 57343
Saturation : 57343
Hue : -8183

**Reply by awawa-dev** [reply]
Had same out of space readings from the OS. Probably there is a bug in the Linux kernel UVC driver that manifest on some platforms (or it's triggered by bad platform DTB configuration).  Just set in HyperHDR: brightness & hue to 1, contrast and saturation to 128 and use custom LUT table for MS2109 for best effect.

**Reply by NiiroX** [reply]
Hi
 Sorry it took so long to answer!!
Big thanks  for your advice, I changed the parameters of the grabber and everything is ok! The colors of the leds have become faithful to the images!!
Thanks again!!
And promised next time I would read your blog better because the solution was there :-)
Respect
